src/com/facebook/buck/android/ApkInstallStep.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2015-present Facebook, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain
+ * a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.facebook.buck.android;
+
+import com.facebook.buck.event.BuckEventBus;
+import com.facebook.buck.rules.InstallableApk;
+import com.facebook.buck.step.ExecutionContext;
+import com.facebook.buck.step.Step;
+import com.facebook.buck.util.Console;
+import com.google.common.base.Optional;
+
+import com.android.ddmlib.IDevice;
+
+public class ApkInstallStep implements Step {
+
+  private final InstallableApk installableApk;
+
+  public ApkInstallStep(InstallableApk apk) {
+    this.installableApk = apk;
+  }
+
+  public ApkInstallStep(AndroidInstrumentationApk apk) {
+    this.installableApk = apk;
+  }
```

Comment:
I think you could replace both of these constructors that just take an `AndroidBinary`, right?